tests/unit/utils/test_urlutils.py

```diff
@@ -787,6 +787,27 @@ def test_hypothesis(self, source):
             assert parsed == source
 
 
+@pytest.mark.parametrize('url, is_pretty, expected', [
+    (QUrl('https://example.com'), False, 'https://example.com'),
+    (QUrl('https://example.com/page'), False, 'https://example.com/page'),
+    (QUrl('ftp://example.com'), False, 'ftp://example.com'),
+    (QUrl('ftp://user:password@example.com'), False, 'ftp://user@example.com'),
+    (QUrl('https://example.com?ref=test'), False, 'https://example.com'),
+    (QUrl('https://example.com?ref=test&example=yes'), False,
+     'https://example.com?example=yes'),
+    (QUrl('https://example.com?ref=test'), False, 'https://example.com'),
```

Comment:
flake8 complains that this is a duplicate test, as it's identical to the one 3 likes above

```suggestion
```